File: 백준/Bronze/5692. 팩토리얼 진법/팩토리얼 진법.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');

const factorials = [1];
for (let i = 1; i <= 5; i++) {
    factorials[i] = factorials[i-1] * i;
}

for (let i = 0; i < input.length; i++) {
    const num = input[i];
       
    if (num === '0') {
        break; 
    }
    
    let answer = 0;
    const reversed = num.split('').reverse().join('');
    reversed.split('').forEach((ch, j) => {
        answer += Number(ch) * factorials[j+1] 
    });
    console.log(answer);
}

File: 백준/Bronze/5692. 팩토리얼 진법/팩토리얼 진법.py
import math
import sys
input = sys.stdin.readline

factorials = [math.factorial(i) for i in range(6)]

while True:
    num = input().strip()
    
    if num == "0":
        break
    
    answer = 0
    for i, ch in enumerate(num[::-1]):
        answer += int(ch) * factorials[i+1] 
        
    print(answer)